File: VB/AuditDemo.Win/WinApplication.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Win

Namespace AuditDemo.Win
	Partial Public Class AuditDemoWindowsFormsApplication
		Inherits WinApplication

		Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
			args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
		End Sub
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub AuditDemoWindowsFormsApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles Me.DatabaseVersionMismatch
			'if(System.Diagnostics.Debugger.IsAttached) {
				e.Updater.Update()
				e.Handled = True
			'}
		End Sub
	End Class
End Namespace

File: CS/AuditDemo.Win/WinApplication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Win;

namespace AuditDemo.Win
{
	public partial class AuditDemoWindowsFormsApplication : WinApplication {
		public AuditDemoWindowsFormsApplication() {
			InitializeComponent();
		}

		private void AuditDemoWindowsFormsApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
			//if(System.Diagnostics.Debugger.IsAttached) {
				e.Updater.Update();
				e.Handled = true;
			//}
		}
	}
}
